v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java

```diff
@@ -156,18 +162,32 @@ Statement databaseOptionsSQL() {
         return Statement.of(
             "SELECT t.option_name, t.option_type, t.option_value "
                 + " FROM information_schema.database_options AS t "
-                + " WHERE t.catalog_name = '' AND t.schema_name = ''");
+                + " WHERE t.schema_name = ''");
       case POSTGRESQL:
         return Statement.of(
             "SELECT t.option_name, t.option_type, t.option_value "
                 + " FROM information_schema.database_options AS t "
-                + " WHERE t.schema_name NOT IN "
-                + "('information_schema', 'spanner_sys', 'pg_catalog')");
+                + " WHERE t.schema_name = 'public'");
       default:
         throw new IllegalArgumentException("Unrecognized dialect: " + dialect);
     }
   }
 
+  private void listSchemas(Ddl.Builder builder) {
+    Statement.Builder queryBuilder =
+        Statement.newBuilder(
+            "SELECT s.schema_name FROM"
+                + " information_schema.schemata AS s WHERE s.effective_timestamp IS NOT NULL");
```

Comment:
Does this contain only customer created schemas? do we need to filter schema_name not in ('information_schema', 'spanner_sys') like in other queries? 

Is this query same for both GSQL and PG dialects?

Reply:
Yep, this will only for customer created schemas. And it is same for both GSQL and PG dialects.